packages/maps/src/index.ts

```diff
@@ -1,54 +1,147 @@
   }
 
-  public instantiateMap(options: { sql?: string, dataset?: string }) {
+  /**
+   * Instantiate a map based on dataset name or sql query, returning a layergroup
+   *
+   * Note: Get layergroup.metadata.tilejson.vector.tiles to consume MVT tiles
+   * @param options
+   */
+  public async instantiateSimpleMap(options: { sql?: string, dataset?: string }) {
     const { sql, dataset } = options;
 
     if (sql === undefined && dataset === undefined) {
       throw new Error('Please provide a dataset or a SQL query');
     }
 
-    const mapParameters = {
+    const mapConfig = {
       layers: [{
         type: 'cartodb',
         options: {
-            sql: sql || `select * from ${dataset}`
+          sql: sql || `select * from ${dataset}`
         }
       }],
       version: '1.3.1'
     };
 
-    const requestOptions = {
+    const layergroup = await this.instantiateMap(mapConfig);
+    return layergroup;
+  }
+
+  private async instantiateMap(mapConfig: any) {
+    let response;
+    try {
+      const payload = JSON.stringify(mapConfig);
+      response = await fetch(this.makeHttpRequest(payload));
+    } catch (error) {
+      throw new Error(`Failed to connect to Maps API with the user ('${this._conf.username}'): ${error}`);
+    }
+
+    const layergroup = await response.json();
+    if (!response.ok) {
+      this.dealWithWindshaftErrors(response, layergroup);
+    }
+
+    return layergroup;
+  }
+
+  private makeHttpRequest(payload: any) {
+    const parameters = [this._conf.auth, this._client, this.encodeParameter('config', payload)];
+    const url = this.generateUrl(this.generateMapsApiUrl(), parameters);
+    if (url.length < REQUEST_GET_MAX_URL_LENGTH) {
+      return this.getRequest(url);
+    }
+
+    return this.postRequest(payload);
+  }
+
+  private dealWithWindshaftErrors(response: any, layergroup: any) {
+    if (response.status === 401) {
+      throw new Error(
+        `Unauthorized access to Maps API: invalid combination of user ('${this._conf._username}') and apiKey ('${this._conf.apiKey}')`
+      );
+    } else if (response.status === 403) {
+      throw new Error(
+        `Unauthorized access to dataset: the provided apiKey('${this._conf.apiKey}') doesn't provide access to the requested data`
+      );
+    }
+    throw new Error(`${JSON.stringify(layergroup.errors)}`);
+  }
+
+  private getRequest(url: string) {
+    return new Request(url, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json'
+      }
+    });
+  }
+
+  private postRequest(payload: any) {
+    const parameters = [this._auth, this._client];
+
+    return new Request(this.generateUrl(this.generateMapsApiUrl(), parameters), {
       method: 'POST',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify(mapParameters)
-    };
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: payload
```

Comment:
We need to be careful with this because it expects it as a stringified JSON and we're not setting the type properly in the parameter definition

Reply:
👌 Parameters declared as proper string